crowbar_framework/app/models/network_service.rb

```diff
@@ -274,15 +274,18 @@ def transition(inst, name, state)
 
     if state == "discovered"
 
-      db = Proposal.where(barclamp: "network", name: inst).first
       role = RoleObject.find_role_by_name "network-config-#{inst}"
-      if NodeObject.find_node_by_name(name).try(:[], "crowbar").try(:[], "admin_node")
-        @logger.info("Admin node transitioning to discovered state.  Adding switch_config role.")
-        result = add_role_to_instance_and_node("network", inst, name, db, role, "switch_config")
-      end
+      result = false
+
+      Proposal.where(barclamp: "network", name: inst).first do |db|
+        if NodeObject.find_node_by_name(name).try(:[], "crowbar").try(:[], "admin_node")
+          @logger.info("Admin node transitioning to discovered state.  Adding switch_config role.")
+          result = add_role_to_instance_and_node("network", inst, name, db, role, "switch_config")
+        end
 
-      @logger.debug("Network transition: make sure that network role is on all nodes: #{name} for #{state}")
-      result = add_role_to_instance_and_node("network", inst, name, db, role, "network")
+        @logger.debug("Network transition: make sure that network role is on all nodes: #{name} for #{state}")
```

Comment:
Metrics/LineLength: Line is too long. [110/100] (https://github.com/bbatsov/ruby-style-guide#80-character-limits)